app/index.php

```diff
@@ -1,6 +1,143 @@
 <?php
+//namespace app;
+session_start();
 
-require_once 'core/App.php';
-require_once 'core/Controller.php';
+//require_once 'core/App.php';
+//require_once 'core/Controller.php';
+//require_once 'controllers/HomeController.php';
+//require_once 'core/Router.php';
+//require_once 'controllers/PostsController.php';
+//require_once 'controllers/CommentsController.php';
+//require_once 'controllers/UsersController.php';
+//require_once 'controllers/AdminController.php';
 
-$app = new App();
+function my_custom_autoloader($class_name) {
+    // Replace namespace separator with directory separator
+    $class_name = str_replace('\\', DIRECTORY_SEPARATOR, $class_name);
+
+    // Remove the "app" prefix from the class name
+    // This assumes your namespace is always "app"
+    if (strpos($class_name, 'app' . DIRECTORY_SEPARATOR) === 0) {
+        $class_name = substr($class_name, strlen('app' . DIRECTORY_SEPARATOR));
+    }
+
+    // Define the base directory for the application
+    $base_dir = __DIR__ . '/'; // Points to /var/www/html/
+
+    // Construct the file path
+    $file = $base_dir . $class_name . '.php';
+
+    // Require the file if it exists
+    if (file_exists($file)) {
+        require_once $file;
+    } else {
+        // Log or display an error if the class file is not found
+        echo "Class file for {$class_name} not found at {$file}\n";
+    }
+}
+
+// Register the autoloader
+spl_autoload_register('my_custom_autoloader');
+
+
+
+use app\controllers\AdminController;
```

Comment:
Use statements should generally be at the top of the file.  You may want to consider moving this, and some of the code below into the `Router` class